bestbook/app.py

```diff
@@ -24,6 +24,8 @@
         '/logout', views.Logout,
         '/api/', views.Index,
         '/api/observations', views.Observations,
+        '/api/users/<username>/observations',views.UserObservations,
+        '/api/books/<bookid>/observations',views.BookObservations,
```

Comment:
We probably want to use a query string that contains the `work_olid` and `edition_olid` for the book observations, instead of including the book ID in the path.  Calls to get a book's observations will be coming from Open Library, which isn't tracking The Best Book On's book IDs.
Using the work and edition OLIDs, we can query for the book ID and then use that to fetch the observations.

Reply:
Right, we probably want that to be `<olid>` instead of `<bookid>` and then in the model, convert `olid` to `bookid` as follows:

1. patron goes to thebestbookon.com/api/observations/books/OL123W and we get the `<olid>` from the URL
2. if 'W' in olid, e.g. `OL123W`, then `book = Book.get(work_olid=olid)` else `book = Book.get(edition_olid=olid)`
3. Now we have the `book.id` that we need for your code

Reply:
it's still called "bookid" in the url but the view code has been updated so bookid is an olid